### PR TITLE
Use JGroups MTLS setup with certmanager generated certificates

### PR DESCRIPTION
**Pull request by ahus1**
Related to: keycloak/keycloak#25703

Until the PR listed above is merged and the code is available in a nightly build, a locally built image will need to be used.

### CODE REVIEW

File: provision/openshift/Taskfile.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ vars:
   KC_HOSTNAME_SUFFIX: '{{default "$(kubectl get route/console -n openshift-console -o jsonpath=\u0027{.spec.host}\u0027 | cut -d . -f 2-)" .KC_HOSTNAME_SUFFIX}}'
   KC_NAMESPACE_PREFIX: '{{default "$(whoami)-" .KC_NAMESPACE_PREFIX}}'
   KC_ADMIN_PASSWORD: '{{default "$(aws secretsmanager get-secret-value --region eu-central-1 --secret-id keycloak-master-password --query SecretString --output text --no-cli-pager || echo admin)" .KC_ADMIN_PASSWORD}}'
+  KC_JGROUPS_TLS: '{{default "true" .KC_ADMIN_PASSWORD}}'
```
Comment (pruivo) [inline provision/openshift/Taskfile.yaml:9]: typo?
```suggestion
  KC_JGROUPS_TLS: '{{default "true" .KC_JGROUPS_TLS}}'
```

Comment (ahus1) [reply provision/openshift/Taskfile.yaml:9]: Nice find!

**Comment by pruivo** [comment]
`Taskfile.yaml` in `rosa-cross-dc` directory needs to be updated too.

My suggestion is to add the following line `--set infinispan.jgroupsTls={{ .KC_JGROUPS_TLS | default false }}` around this line: 
https://github.com/keycloak/keycloak-benchmark/blob/9451873f69b244c4825d6b164783d376bff54668/provision/rosa-cross-dc/Taskfile.yaml#L280

**Comment by pruivo** [comment]
I've tested this using `openssl s_client -connect <keycloak-pod>:<jgroups-port>` and it shows a certificate:

```
CN = keycloak-jgroups
Requested Signature Algorithms: ECDSA+SHA256:ECDSA+SHA384:ECDSA+SHA512:Ed25519:Ed448:RSA-PSS+SHA256:RSA-PSS+SHA384:RSA-PSS+SHA512:RSA-PSS+SHA256:RSA-PSS+SHA384:RSA-PSS+SHA512:RSA+SHA256:RSA+SHA384:RSA+SHA512
Shared Requested Signature Algorithms: ECDSA+SHA256:ECDSA+SHA384:ECDSA+SHA512:Ed25519:Ed448:RSA-PSS+SHA256:RSA-PSS+SHA384:RSA-PSS+SHA512:RSA-PSS+SHA256:RSA-PSS+SHA384:RSA-PSS+SHA512:RSA+SHA256:RSA+SHA384:RSA+SHA512
Peer signing digest: SHA256
Peer signature type: RSA-PSS
Server Temp Key: X25519, 253 bits
```

The JGroups view was formed properly too.

**Comment by ahus1** [comment]
I've incorporated your suggestions.

> The JGroups view was formed properly too.

Could you think of a good check to add to a readiness probe where we could detect a mis-configuration? 
Something like "the discovery returned some IP addresses, but I wasn't able to connect to them"? It should be a check that would still succeed when there is just a single node, but that would fail once there are two nodes and they are unable to connect to each other especially in situations where the encryption can't be established.

File: provision/rosa-cross-dc/Taskfile.yaml [hunk 2]
```diff
@@ -17,6 +17,7 @@ vars:
   BENCHMARK_DIR: "{{.ROOT_DIR}}/../../benchmark/src/main/content/bin"
   KEYCLOAK_MASTER_PASSWORD:
     sh: aws secretsmanager get-secret-value --region eu-central-1 --secret-id keycloak-master-password --query SecretString --output text --no-cli-pager
+  KC_JGROUPS_TLS: '{{default "true" .KC_JGROUPS_TLS}}'
```
Comment (pruivo) [inline provision/rosa-cross-dc/Taskfile.yaml:20]: The global variable is not required. If you want to keep it, the line 280 can be simplified:
`--set infinispan.jgroupsTls={{ .KC_JGROUPS_TLS }}`

Comment (ahus1) [reply provision/rosa-cross-dc/Taskfile.yaml:20]: Thanks. I would like to keep it as it serves as some kind of reference at the top of the file. And I'd like to keep it `true`. 

Simplified the setup below as you suggested.

**Review by pruivo — APPROVED** [review]

**Comment by ahus1** [comment]
> Could you think of a good check to add to a readiness probe where we could detect a mis-configuration?

Created a follow-up issue #692

**Pull request merged**